### ISSUE TITLE
Transfers: intermittent abort upload test failure

### ISSUE THREAD

**Issue by JurgenR** [JurgenR/aioslsk#129]
The test for aborting an upload (`test_transfer_abortUpload`) is failing intermittently (about 1/30). This is likely some kind of timing issue that needs to be fixed.

As a start enable logging for the tests to get a better insight into what's happening (however there is a test that fails when logging is enabled which needs to be fixed first)

See https://github.com/JurgenR/aioslsk/actions/runs/6857046661/job/18645400690

**Comment by JurgenR** [comment]
Might be related to #122

**Comment by JurgenR** [comment]
This ticket might be resolved by PR #132 . Will keep it open for a bit

**Comment by JurgenR** [comment]
Tests look stable after numerous runs, closing

**Comment by JurgenR** [comment]
Re-opening already: https://github.com/JurgenR/aioslsk/actions/runs/6949910579/job/18908995959?pr=140

**Comment by JurgenR** [comment]
This was for the abort download test instead of the abort upload test. The logs are interesting:

```
2023-11-21T21:46:53.2000236Z 2023-11-21 21:46:53 [   DEBUG] transitioning transfer state from INITIALIZING to UPLOADING (model.py:164)
2023-11-21T21:46:53.2003885Z 2023-11-21 21:46:53 [   DEBUG] 127.0.0.1:50732 setting state to PeerConnectionState.TRANSFERRING : PeerConnection(hostname='127.0.0.1', port=50732, state=ConnectionState.CONNECTED, incoming=True, obfuscated=False, username='user1', connection_state=PeerConnectionState.AWAITING_TICKET, connection_type='F') (connection.py:517)
2023-11-21T21:46:53.2006024Z 2023-11-21 21:46:53 [   DEBUG] transitioning transfer state from INITIALIZING to DOWNLOADING (model.py:164)
2023-11-21T21:47:47.1011184Z 2023-11-21 21:47:47 [   DEBUG] 127.0.0.1:50732 : disconnecting : REQUESTED (connection.py:236)
2023-11-21T21:47:47.1012987Z 2023-11-21 21:47:47 [   DEBUG] transitioning transfer state from DOWNLOADING to COMPLETE (model.py:164)
2023-11-21T21:47:47.1015681Z 2023-11-21 21:47:47 [   DEBUG] localhost:4444 : send message : RemoveUser.Request(username='user1') (connection.py:428)
2023-11-21T21:47:47.1018282Z 2023-11-21 21:47:47 [    INFO] 127.0.0.1:32790: receive : RemoveUser.Request(username='user1') (peer.py:97)
2023-11-21T21:47:47.1021024Z 2023-11-21 21:47:47 [   DEBUG] transitioning transfer state from COMPLETE to ABORTED (model.py:164)
2023-11-21T21:47:47.1418721Z 2023-11-21 21:47:47 [   DEBUG] transitioning transfer state from UPLOADING to COMPLETE (model.py:164)
```

* It doesn't seem like an attempt is made to abort until the very end
* The transition from complete to abort should not be possible

**Comment by JurgenR** [comment]
The download speed is limited to 5kbps, while the upload speed is 4kbps. There might be a tiny chance control is never given back to the event loop because the asyncio.sleep is never called in the rate limiter? I believe this to be unluckily though as 2 IO operations are done (reading from socket and writing to file)

**Comment by JurgenR** [comment]
Another thought was that the `await self._writer.wait_closed()` call hangs. But there should've been a log message indicating the connection is closing before getting to that point which isn't seen

**Comment by JurgenR** [comment]
Another one for upload: https://github.com/JurgenR/aioslsk/actions/runs/6964121963/job/18950872996

**Comment by JurgenR** [comment]
Probably related to : https://bugs.python.org/issue42130

When using `asyncio.wait_for` the `CancelledError` will sometimes get swallowed. I wrote a script to reproduce locally and it's swallowed about 50% of the time. This percentage doesn't really match with how much the abort error occurs but the code only uses `wait_for` when reading/writing from the network, if it's cancelled during writing/reading from files it should be fine.

This comment : `https://github.com/python/cpython/issues/107547#issuecomment-1661568521` suggests that instead of `wait_for` the `asyncio.timeout` should be used but this was only introduced in Python 3.11. It's also implied that the fix for this might only be introduced in 3.12.

There is a library with the same behaviour as `asyncio.timeout` here: https://github.com/aio-libs/async-timeout . But there should be some investigation whether this resolves the abort issue seen here or not. If it does work then the `asyncio.wait_for` call really be replaced everywhere as there might be other similar issues lurking.

**Comment by JurgenR** [comment]
Tried the library out on the script that initially had a 50% failure rate and everything was cancelled successfully
